test/Microsoft.IdentityModel.Tokens.Tests/CryptoProviderCacheTests.cs

```diff
@@ -426,6 +427,73 @@ public static TheoryData<CryptoProviderCacheTheoryData> TryGetSignatureProviderT
             }
         }
 
+        /// <summary>
+        /// Checks that the cache and any crypto providers contained inside of it are properly garbage collected.
+        /// </summary>
+        [Fact]
+        public void CryptoProviderCacheDisposal()
+        {
+            TestUtilities.WriteHeader($"{this}.CryptoProviderCacheDisposal");
+            var context = new CompareContext();
+            WeakReference cacheReference = null;
```

Comment:
This mostly works, but it would be better to test this from the perspective of how we expect it to be used.

e.g. If we make these regular references and test _disposed and DisposeCalled.

Debugging this, looks like Dispose is not called on  cryptoProviderReference

Reply:
That makes sense. I think the tests checking for garbage collection don't really have a place here.

I've gone ahead and simplified the tests to check that DisposeCalled is properly set on the InMemoryCryptoProviderCache.

It makes sense that Dispose() wouldn't be called on the cryptoProviderReference, since that's something that would need to be manually done by the user...